staging/src/k8s.io/kubectl/pkg/generate/versioned/run.go

```diff
@@ -178,14 +178,14 @@ func updatePodPorts(params map[string]string, podSpec *v1.PodSpec) (err error) {
 	port := -1
 	hostPort := -1
 	if len(params["port"]) > 0 {
-		port, err = strconv.Atoi(params["port"])
+		port, err = utilsnet.ParsePort(params["port"], false)
 		if err != nil {
 			return err
 		}
 	}
 
 	if len(params["hostport"]) > 0 {
-		hostPort, err = strconv.Atoi(params["hostport"])
+		hostPort, err = utilsnet.ParsePort(params["hostport"], false)
```

Comment:
is 0 allowed? 
looking the if below it seems so, thus we should use `true`
`if hostPort > 0 && port < 0 {`

Reply:
https://github.com/kubernetes/kubernetes/blob/ede025af1b54ee20c0abbce553a5b15ef31225f8/staging/src/k8s.io/api/core/v1/types.go#L1749-L1757
I see that the port limit is 0 to 65536 as shown above, so I set it to false.

Reply:
@liggitt if theses values `HostPort `, `ContainerPort` are not specified ... don't they take the zero value of the type? in this case 0